backend/src/main/java/com/my/shope/backend/product/ProductService.java

```diff
@@ -145,12 +149,61 @@ public void setProductDetails(Product product,String textFilePath) throws IOExce
     }
 
     public void deleteProduct(String productId){
+        removeFromShoppingCartOrOrderedByExecuted(true,productId);
+        removeFromShoppingCartOrOrderedByExecuted(false,productId);
         Product product = getProductById(productId);
         List<String> imageIds = product.getImageIDs();
         productRepo.delete(product);
         fileService.deleteImagesByIds(imageIds);
+    }
+
+
+    public Product updateProduct(String productId, MultipartFile[] multipartFile) throws IOException {
+        Product productToUpdate = getProductById(productId);
 
+        // if only product_details in the multipartFile then update only the details
+        if(multipartFile.length == 1 && Objects.requireNonNull(multipartFile[0].getOriginalFilename()).startsWith("product_details")){
+            fileService.saveProductDetailsFile(multipartFile[0]);
+            setProductDetails(productToUpdate, DETAILS_PATH);
+            return productRepo.save(productToUpdate);
+        }
+
+        // case contains the product_details file and photos
+        for (MultipartFile file : multipartFile) {
+            if (Objects.requireNonNull(file.getOriginalFilename()).startsWith("product_details")) {
+                fileService.saveProductDetailsFile(file);
+                setProductDetails(productToUpdate, DETAILS_PATH);
+            } else {
+                fileService.deleteImagesByIds(productToUpdate.getImageIDs());
+            }
+
+        }
+
+        //just to delete old photos and sett the new photos to mey productToUpdate
+        productToUpdate.setImageIDs(new ArrayList<>());
+        for (MultipartFile file : multipartFile) {
+            if (!Objects.requireNonNull(file.getOriginalFilename()).startsWith("product_details")) {
+                productToUpdate.getImageIDs().add(fileService.saveFile(file).getId());
+            }
+        }
+        System.out.println(productToUpdate);
+        return productRepo.save(productToUpdate);
```

Comment:
Diese Methode sind unnötig kompliziert aus. 
Kann man mit ein bißchen Umstrukturierung die ganzen Fälle nicht in einem `for (MultipartFile file: multipartFile)` abbilden?

Reply:
hmmm das weiss ich nicht ich muss noch mal schauen :)